src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -325,6 +346,49 @@ public Task<IAsyncCursor<RawBsonDocument>> ResumeAsync(IReadBinding binding, Can
             return aggregateOperation.ExecuteAsync(binding, cancellationToken);
         }
 
+        /// <inheritdoc />
+        public void ReviewResumeParams()
+        {
+            // Docs: if the ChangeStream has a cached postBatchResumeToken and has returned all documents in the most recent batch(or the most recent batch was empty):
+            // The driver MUST set resumeAfter to the cached postBatchResumeToken.
+            // The driver MUST NOT set startAfter. If startAfter was in the original aggregation command, the driver MUST remove it.
+            // The driver MUST NOT set startAtOperationTime. If startAtOperationTime was in the original aggregation command, the driver MUST remove it.
+            // todo: should be implemented in scope of the CSHARP-2514.
+
+            // Docs: if the ChangeStream has a cached documentResumeToken:
+            // The driver MUST set resumeAfter to the cached documentResumeToken.
+            // The driver MUST NOT set startAfter. If startAfter was in the original aggregation command, the driver MUST remove it.
+            // The driver MUST NOT set startAtOperationTime. If startAtOperationTime was in the original aggregation command, the driver MUST remove it.
+            // todo:
```

Comment:
Is this condition the same with the condition in the 378 line below? 
(At least until we will have implemented the ticket 2514).

Reply:
I don't understand the question. Given that this method should go away anyway I would rephrase the question after refactoring.